monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -545,28 +543,38 @@ def get_issues():
         logger.info("Issues generated for reporting")
         return issues_dict
 
-    @staticmethod
-    def is_latest_report_exists():
+    @classmethod
+    def get_latest_event_timestamp(cls) -> float:
+        if not cls._agent_event_repository:
+            raise RuntimeError("Agent event repository does not exist")
+
+        # TODO: Add `get_latest_event` to the IAgentEventRepository
+        agent_events = cls._agent_event_repository.get_events()
+        latest_timestamp = max(agent_events, key=lambda event: event.timestamp).timestamp
+
+        return latest_timestamp
+
+    @classmethod
+    def is_latest_report_exists(cls):
         """
         This function checks if a monkey report was already generated and if it's the latest one.
         :return: True if report is the latest one, False if there isn't a report or its not the
         latest.
         """
-        latest_report_doc = mongo.db.report.find_one({}, {"meta_info.latest_monkey_modifytime": 1})
 
-        if latest_report_doc:
-            report_latest_modifytime = latest_report_doc["meta_info"]["latest_monkey_modifytime"]
-            latest_monkey_modifytime = Monkey.get_latest_modifytime()
-            return report_latest_modifytime == latest_monkey_modifytime
+        if cls._report:
+            report_latest_event_timestamp = cls._report["meta_info"]["latest_event_timestamp"]
+            latest_event_timestamp = cls.get_latest_event_timestamp()
+            return report_latest_event_timestamp == latest_event_timestamp
 
         return False
 
-    @staticmethod
-    def get_report():
-        if not ReportService._agent_repository.get_agents():
+    @classmethod
+    def get_report(cls):
+        if not cls._agent_repository.get_agents():
             return {}
```

Comment:
You don't need this check if the default report is `{}`

Reply:
Why? This check is returning empty if we don't have any agents in the repository.

Reply:
It's not readable and unnecessary. `is_report_outdated` checks latest event timestamp. So if there are no events, the default (`{}`) is the latest and we can return it